e2e/test/scenarios/actions/actions-on-dashboards.cy.spec.js

```diff
@@ -277,6 +282,113 @@ const MODEL_NAME = "Test Action Model";
           });
         });
 
+        it("hide actions in public dashboards (metabase#34395)", () => {
+          const dashboardDetails = {
+            name: "Public Dashboard",
+          };
+          cy.get("@modelId")
+            .then(id => {
+              createImplicitAction({
+                kind: "create",
+                model_id: id,
+              });
+            })
+            .then(({ body: action }) => {
+              cy.createDashboard(dashboardDetails).then(
+                ({ body: dashboard }) => {
+                  updateDashboardCards({
+                    dashboard_id: dashboard.id,
+                    cards: [
+                      getActionCardDetails({
+                        action_id: action.id,
+                        label: "Create",
+                      }),
+                    ],
+                  });
+                  cy.visit(`/dashboard/${dashboard.id}`);
+                  cy.wrap(dashboard.id).as("dashboardId");
+                },
+              );
+            });
+
+          cy.log("The action should be visible in the dashboard");
+          cy.findByRole("button", { name: "Create" }).should("be.visible");
+
+          cy.log("Visit public dashboard");
+          cy.get("@dashboardId")
+            .then(dashboardId => {
+              createPublicDashboardLink(dashboardId);
+            })
+            .then(({ body: { uuid } }) => {
+              cy.visit(`/public/dashboard/${uuid}`);
+            });
+
+          cy.log("Assert public dashboard");
+          cy.findByRole("heading", { name: dashboardDetails.name }).should(
+            "be.visible",
+          );
+          /**
+           * cy.button("Create") doesn't work because the label is
+           * `Actions are not enabled for this database` for disabled actions
+           */
+          cy.findByRole("main").findByText("Create").should("not.exist");
```

Comment:
I was a bit surprised using `cy.button("Create")` doesn't fail the test before introducing the fix. Turned out the action label is different when it's disabled.